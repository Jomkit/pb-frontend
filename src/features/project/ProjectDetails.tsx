import { useParams } from "react-router-dom"
import Card from "../../components/Card"
import Frame from "../../components/Frame"
import { useEffect, useState } from "react";
import Api from "../../api";
import TaskList from "../task/TaskList";
import useLocalStorage from "../../components/hooks/useLocalStorage";

const ProjectDetails = () => {
  const {projectId} = useParams();
  const [currProjectId, setCurrProjectId] = useLocalStorage("currProjectId");
  const [project, setProject] = useState({} as any);
  const [tasks, setTasks] = useState([] as any);
  const [isLoading, setIsLoading] = useState(true);
  
  useEffect(() => {
    const getProject = async () => {
      const result = await Api.getProject(projectId!);
      const tasks = await Api.findProjectTasks(projectId!);
      console.log("tasks", tasks);
      setTasks(tasks);
      setProject(result);
      setCurrProjectId(projectId);
      setIsLoading(false);
    }
    getProject();
    console.log("currProjectId", currProjectId);
    
  }, []);
  
  return (
    <Frame bgColor="bg-cyan-400" title="">
      {isLoading ? <p>Loading...</p> : 
      <>
        <Card title={project.name} subtitle={project.note}>
            <TaskList tasks={tasks} loading={isLoading}/>
        </Card>
      </>}
    </Frame>
    
  )
}

export default ProjectDetails