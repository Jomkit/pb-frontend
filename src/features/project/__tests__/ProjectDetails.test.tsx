import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import ProjectDetails from "../ProjectDetails"

vi.mock("../../../api");

it("should render without crashing", () => {
    render(
        <MemoryRouter initialEntries={["/projects/1"]}>
            <ProjectDetails />
        </MemoryRouter>
    )
})


it("should match snapshot of loading", () => {
    const { asFragment } = render(
        <MemoryRouter initialEntries={["/projects/1"]}>
            <ProjectDetails />
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})  

it("should match snapshot after loading complete", async () => {
    const { asFragment, findAllByText } = render(
        <MemoryRouter initialEntries={["/projects/1"]}>
            <ProjectDetails />
        </MemoryRouter>
    )

    const allTestTasks = await findAllByText("testTask", { exact: false });
    expect(allTestTasks[0]).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
})
