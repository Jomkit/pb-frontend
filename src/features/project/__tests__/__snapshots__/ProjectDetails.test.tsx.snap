// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot after loading complete 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-cyan-400"
  >
    <h1 />
    <div
      class="p-10 justify-center"
    >
      <div
        class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
      >
        <div
          class="text-center"
        >
          <h2 />
          <h4 />
        </div>
        <div
          class="w-3/4 m-auto"
        >
          <div
            class="text-start"
          >
            <button
              class="btn bg-green-400 border-solid border-2 border-black"
            >
              New Task
            </button>
          </div>
          <div
            class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
          >
            <div
              class="text-center"
            >
              <h2>
                testTask 1
              </h2>
              <h4>
                Project id: Unassigned
              </h4>
            </div>
            <a
              href="/tasks/1"
            >
              <p
                class="btn bg-blue-500 text-white"
              >
                Click for more info
              </p>
            </a>
            <div
              class="m-2"
            >
              <div
                class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
              >
                <div
                  class="text-center"
                >
                  <h2 />
                  <h4 />
                </div>
                <h2
                  class="text-start"
                >
                  Timer
                </h2>
                <div
                  class="flex justify-between"
                >
                  <div
                    class="text-center"
                  >
                    <h3
                      class="text-2xl bg-slate-200 w-fit m-auto px-4 py-1 rounded border-2 border-slate-500"
                    >
                      25:00
                    </h3>
                  </div>
                  <div
                    id="set-timer"
                  >
                    <button
                      class="btn bg-green-400"
                      id="start-btn"
                    >
                      <i
                        class="fa-solid fa-play"
                      />
                    </button>
                    <button
                      class="btn bg-red-500"
                      id="restart-btn"
                    >
                      <i
                        class="fa-solid fa-rotate-right"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
          >
            <div
              class="text-center"
            >
              <h2>
                testTask 2
              </h2>
              <h4>
                Project id: Unassigned
              </h4>
            </div>
            <a
              href="/tasks/2"
            >
              <p
                class="btn bg-blue-500 text-white"
              >
                Click for more info
              </p>
            </a>
            <div
              class="m-2"
            >
              <div
                class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
              >
                <div
                  class="text-center"
                >
                  <h2 />
                  <h4 />
                </div>
                <h2
                  class="text-start"
                >
                  Timer
                </h2>
                <div
                  class="flex justify-between"
                >
                  <div
                    class="text-center"
                  >
                    <h3
                      class="text-2xl bg-slate-200 w-fit m-auto px-4 py-1 rounded border-2 border-slate-500"
                    >
                      25:00
                    </h3>
                  </div>
                  <div
                    id="set-timer"
                  >
                    <button
                      class="btn bg-green-400"
                      id="start-btn"
                    >
                      <i
                        class="fa-solid fa-play"
                      />
                    </button>
                    <button
                      class="btn bg-red-500"
                      id="restart-btn"
                    >
                      <i
                        class="fa-solid fa-rotate-right"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`should match snapshot of loading 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-cyan-400"
  >
    <h1 />
    <div
      class="p-10 justify-center"
    >
      <p>
        Loading...
      </p>
    </div>
  </section>
</DocumentFragment>
`;
